src/main/kotlin/org/jetbrains/research/testspark/actions/llm/LLMSampleSelectorFactory.kt

```diff
@@ -172,4 +180,55 @@ class LLMSampleSelectorFactory(private val project: Project) : PanelFactory {
             testSamplePanelFactory.enabledComponents(isEnabled)
         }
     }
+
+    /**
+     * Retrieves a list of test samples from the given project.
+     *
+     * @return A list of strings, representing the names of the test samples.
+     */
+    private fun collectTestSamples(project: Project, testNames: MutableList<String>, initialTestCodes: MutableList<String>) {
+        val projectFileIndex: ProjectFileIndex = ProjectRootManager.getInstance(project).fileIndex
+        val javaFileType: FileType = FileTypeManager.getInstance().getFileTypeByExtension("java")
+
+        projectFileIndex.iterateContent { file ->
+            if (file.fileType === javaFileType) {
+                try {
+                    val psiJavaFile = (PsiManager.getInstance(project).findFile(file) as PsiJavaFile)
+                    val psiClass = psiJavaFile.classes[
+                        psiJavaFile.classes.stream().map { it.name }.toArray()
+                            .indexOf(psiJavaFile.name.removeSuffix(".java")),
+                    ]
+                    var imports = psiJavaFile.importList?.allImportStatements?.map { it.text }?.toList()
+                        ?.joinToString("\n") ?: ""
+                    if (psiClass.qualifiedName != null && psiClass.qualifiedName!!.contains(".")) {
+                        imports += "\nimport ${psiClass.qualifiedName?.substringBeforeLast(".") + ".*"};"
+                    }
+                    psiClass.allMethods.forEach { method ->
+                        val annotations = method.modifierList.annotations
+                        annotations.forEach { annotation ->
+                            if (annotation.qualifiedName == "org.junit.jupiter.api.Test" || annotation.qualifiedName == "org.junit.Test") {
+                                val code: String = createTestSampleClass(imports, method.text)
+                                testNames.add(createMethodName(psiClass, method))
+                                initialTestCodes.add(code)
+                            }
+                        }
+                    }
+                } catch (_: Exception) {
+                }
+            }
+            true
+        }
+    }
+
+    private fun createTestSampleClass(imports: String, methodCode: String): String {
+        var normalizedImports = imports
+        if (normalizedImports.isNotBlank()) normalizedImports += "\n\n"
+        return normalizedImports +
+            "public class TestSample {\n" +
+            "   $methodCode\n" +
+            "}"
+    }
+
+    private fun createMethodName(psiClass: PsiClass, method: PsiMethod): String =
+        "<html>${psiClass.qualifiedName}#${method.name}</html>"
```

Comment:
[Nitpick] I would rename the method to `createUIPresantableMethodName`. First, I thought this function would create a name akin to a method signature, not HTML.

Maybe this function knows too much about PSI components. How about the following?
```kotlin
private fun createUIPresantableMethodName(classQualifiedName: String, methodName: String): String =
        "<html>${classQualifiedName}#${methodName}</html>"

// ...

val uiMethodName = createUIPresantableMethodName(psiClass.qualifiedName, psiMethod.name)
```